narwhals/_polars/expr.py

```diff
@@ -400,6 +400,22 @@ class PolarsExprDateTimeNamespace(
 class PolarsExprStringNamespace(
     PolarsExprNamespace, PolarsStringNamespace[PolarsExpr, pl.Expr]
 ):
+    def to_titlecase(self) -> PolarsExpr:
+        native_expr = self.native
+        backend_version = self.compliant._backend_version
+
+        if backend_version < (1, 5):  # pragma: no cover
+            native_result = (
+                native_expr.str.to_lowercase()
+                .str.extract_all(r"[a-z0-9]+[^A-Za-z0-9]*")
+                .list.eval(pl.element().str.to_titlecase())
```

Comment:
At this point, this operation is "safe", since the non-alphanumeric chars are all at the end of the string